core/blockchain.go

```diff
@@ -1610,8 +1623,8 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 	// First block is pruned, insert as sidechain and reorg only if TD grows enough
 	case err == consensus.ErrPrunedAncestor:
```

Comment:
when can this happen?

Reply:
This happens when it tries to verify the block header's transaction, it has the header but not the state.
To be honest, I didn't know if it's possible to not have the state of a really old block, but what I did is to cut the chain insertion in that moment.
That case was calling to:
`return bc.insertSideChain(block, it)`
which I removed, but, you are right, maybe I could still throw the error, and not just cut the iteration